code/Util/ListExtensions.cs

```diff
@@ -16,7 +16,7 @@ public static void Shuffle<T>( this IList<T> list )
 		while ( n > 1 )
```

Comment:
Wait shouldn't this be `n > 0`? Might be why it didn't originally work...

Reply:
No.
https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle#The_modern_algorithm